backend/src/main/java/com/benine/backend/preset/autopresetcreation/AutoPresetCreator.java

```diff
@@ -15,62 +15,68 @@
 import java.util.concurrent.TimeoutException;
 
 
+
 /**
- * Abstract class used for autocreating presets.
+ * Abstract class used for auto-creating presets.
  */
 public abstract class AutoPresetCreator {
 
   private static long timeout = 5000;
   private PresetController presetController;
+  
   /**
    * Creates a new AutoPresetCreator object.
    * @param presetController the presetController to add the created presets too.
    */
   public AutoPresetCreator(PresetController presetController) {
     this.presetController = presetController;
   }
+ 
   /**
    * Automatically creates presets for the selected camera.
    * Calls generatePositions to get the positions of the presets.
    * Automatically adds the presets to the presetcontroller.
    * @param cam the camera to create presets for.
-   * @param subViews the subviews to generate positions for.
-   * @return A collection of the created presets.
+   * @param subViews the sub-views to generate positions for.
+   * @return A collection of the id's of the created presets.
    * @throws CameraConnectionException when camera cannot be reached.
-   * @throws InterruptedException when interupted while waiting for cam to move.
+   * @throws InterruptedException when interrupted while waiting for cam to move.
    * @throws TimeoutException if the camera is moving too slow or not at all.
    * @throws CameraBusyException if the camera is busy.
    * @throws IOException if exception occurs when creating the preset image.
    * @throws StreamNotAvailableException if a camera stream cannot be reached.
    * @throws SQLException if exception occurs while writing to the database.
    */
-  public Collection<IPCameraPreset> createPresets(IPCamera cam, Collection<SubView> subViews)
+  public ArrayList<Integer> createPresets(IPCamera cam, Collection<SubView> subViews)
           throws CameraConnectionException, CameraBusyException, InterruptedException,
           TimeoutException, IOException, StreamNotAvailableException, SQLException {
     if (cam.isBusy()) {
       throw new CameraBusyException("The camera is busy.", cam.getId());
     }
     cam.setBusy(true);
     ArrayList<IPCameraPreset> presets = new ArrayList<>();
+    ArrayList<Integer> presetIDs = new ArrayList<>();
 
     cam.setBusy(false);
     cam.setAutoFocusOn(true);
     cam.setBusy(true);
     for (ZoomPosition pos : generatePositions(cam, subViews)) {
-      presets.add(generatePresetFromPos(pos, cam));
+      IPCameraPreset currentPreset = generatePresetFromPos(pos,cam);
+      presetIDs.add(currentPreset.getId());
+      presets.add(currentPreset);
     }
     cam.setBusy(false);
-    return presets;
+    return presetIDs;
```

Comment:
I don't think its a good idea to return an list of presetids instead of a collection of preset here since this won't be useful when implementing the progress bar functionality.